### PR TITLE
Use bot token for issue workflows and increase stale issue budget

### PR DESCRIPTION
**Pull request by lnkuiper**
The stale issue bot timed out after replying to 30 issues, so I've increased its budget (for now). Maybe we can decrease it later when the bot has gone through all of the stale issues.

### CODE REVIEW

**Comment by carlopi** [comment]
I would have IssuesCloseStale.yml using the repository_dispatch mechanism used by the other workflows, at that point it will automatically be externally triggered every night (around 2am Amsterdam time, see https://github.com/duckdb/duckdb-workflow-trigger/blob/main/.github/workflows/Workflow-Trigger.yml) BUT someone forking duckdb will not have to be bothered by those extra scheduled wofklows (I don't think there is any way to stop them).

And then mirroring the other workflows I think it might be somehow useful (or at least not damaging) to have the extra option to trigger this manually via workflow_dispatch.

Basically I would do:
```bash
+      repository_dispatch:
+      workflow_dispatch:
-      schedule:
-       - cron: '00 0 * * *'
```

**Comment by lnkuiper** [comment]
@carlopi Do we want these on the other workflows as well then? The ones that create mirror issues? I don't think we want to run those on forks either. I'm happy to add these keywords, but I'm not sure how they work

**Comment by carlopi** [comment]
> > @carlopi Do we want these on the other workflows as well then? The ones that create mirror issues? I don't think we want to run those on forks either. I'm happy to add these keywords, but I'm not sure how they work
> 
> It's not a problem for CI runs that trigger on issue creation because you can't create issues in forks. It's mainly a problem for scheduled cron jobs

This, basically it's not super clean, but they will most likely not run into that / I am not sure how to do that. I would solve only scheduled that's problematic since even without any action (say you fork and forget about it), it will keep being executed.

`repository_dispatch` = you can externally send a generic signal to a repository that translates to "execute all workflows that have a repository_dispatch trigger". We use that from this external workflow: https://github.com/duckdb/duckdb-workflow-trigger/tree/main/.github/workflows (that has the right privileges to do so).

`workflow_dispatch` = the same, but it targets a specific workflow + with the right privileges you get a button to activate it from the WebUI.

**Review by carlopi — APPROVED** [review]
